MyHashMap.java

```diff
@@ -0,0 +1,228 @@
+
+        }
+    }
+
+    private void resize() {
+        int size = this.size * 2;
+        Node[] tempNode = new MyHashMap.Node[size];
+        System.arraycopy(nodesArr, 0, tempNode, 0, this.size);
+        this.nodesArr = tempNode;
+        this.size = size;
+        this.threshold = (int) (this.size * loadFactor);
+    }
+
+    @Override
+    public V remove(K key) {
+        if (key == null) {
+            if (nodesArr[0].key == null) {
+                Node currNode = nodesArr[0];
+                nodesArr[0] = currNode.next;
+                return currNode.value;
+            } else {
+                if (nodesArr[0].next != null) {
+                    Node currNode = nodesArr[0].next;
+                    if (currNode.key == null) {
+                        nodesArr[0].next = currNode.next;
+                        return currNode.value;
+                    }
+                    Node lastNode;
+                    while (currNode.key != null) {
+                        lastNode = currNode;
+                        currNode = lastNode.next;
+                        if (nodesArr[0].key == null) {
+                            lastNode.next = currNode.next;
+                            currNode = currNode.next;
+                        }
+                    }
+                }
+            }
+
+        } else {
+            int sizeArr = 16;
+            while (sizeArr <= size) {
+                int index = hashIndex(key, sizeArr);
+                if (nodesArr[index] != null) {
+                    if (nodesArr[index].key.equals(key)) {
+                        Node currNode = nodesArr[index];
+                        nodesArr[index] = nodesArr[index].next;
+                        return currNode.value;
+                    } else {
+                        Node lastNode = nodesArr[index];
+                        Node currNode = nodesArr[index].next;
+                        if (currNode.key.equals(key)) {
+                            lastNode.next = currNode.next;
+                            return currNode.value;
+                        }
+                        while (currNode.next != null) {
+                            lastNode = currNode;
+                            currNode = lastNode.next;
+                            if (nodesArr[index].key.equals(key)) {
+                                lastNode.next = currNode.next;
+                                return currNode.value;
+                            }
+                        }
+                    }
+                }
+                sizeArr *= 2;
+            }
+        }
+        return null;
+    }
+
+    @Override
+    public void clear() {
+        loadFactor = 0.75;
+        threshold = 12;
+        size = 16;
+        capacity = 0;
+        nodesArr = new MyHashMap.Node[size];
+    }
+
+    @Override
+    public int size() {
+        return size;
+    }
+
+    @Override
+    public V get(K key) {
+        if (key == null) {

```

Comment:
I would not recommend you to allow null key, it makes your code more complicated, and actually does not makes much sense

Reply:
> I would not recommend you to allow null key, it makes your code more complicated, and actually does not makes much sense

I read articles on habr, there was such opportunity therefore I realized it, actually it took a lot of time :) though if honestly, I too don't see in it sense

Reply:
No problem, your code is more complex and you need just to do more job to make it decent and working